py-polars/polars/lazyframe/frame.py

```diff
@@ -1872,7 +1876,7 @@ def collect(
         collapse_joins: bool = True,
         no_optimization: bool = False,
         streaming: bool = False,
-        engine: EngineType = "cpu",
+        engine: None | EngineType = "cpu",
```

Comment:
Default to `None` here, Document what `None` means in the docstring (`get_engine_affinity` is used)?

```suggestion
        engine: EngineType | None = None,
```

I think we want `None` to be the default, otherwise the `engine_affinity` env var will only be respected when calling `collect(engine=None)`, rather than just `collect()`.